src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageAccountProvider.cs

```diff
@@ -15,6 +16,7 @@ internal class DefaultStorageAccountProvider : IStorageAccountProvider
         private readonly IStorageCredentialsValidator _storageCredentialsValidator;
         private readonly IStorageAccountParser _storageAccountParser;
         private readonly IServiceProvider _services;
+        private readonly ConcurrentDictionary<string, Task<IStorageAccount>> _accounts = new ConcurrentDictionary<string, Task<IStorageAccount>>();
```

Comment:
Intended as Task<IStorageAccount>, because it made for the cleanest change (Dict.GetOrAdd doesn't support async)

Do you see any negatives to Task<IStorageAccount>? Perhaps if you have a faulted task, but as we throw anyway if storage account isn't created I don't see that as an issue.